src/test/java/bio/terra/workspace/common/fixtures/ControlledResourceFixtures.java

```diff
@@ -92,4 +93,24 @@ public static ControlledGcsBucketResource makeControlledGcsBucketResource(UUID w
         ManagedByType.MANAGED_BY_USER,
         BUCKET_NAME);
   }
+
+  /**
+   * Returns a {@link ControlledAiNotebookInstanceResource.Builder} that is ready to be built.
+   *
+   * <p>Tests should not rely on any particular value for the fields returned by this function and
```

Comment:
Interesting. The other methods in this class are being relied on, but I think I made public static constants for some of the values as well.

Reply:
Yep. I've found default builders to be a very helpful testing pattern, whereas many factory function fixtures gets verbose. It's also nice to asks tests to not rely on defaults unless you really want to commit to making them public and modifying all potential uses. Better to ask tests to explicitly set what they're relying on (not that developers always follow instructions.)